nima_core/storage/unified_db.py

```
@@ -102,7 +103,18 @@ def _writer(fn):
-            return fn(conn, *args, **kwargs)
+            result = fn(conn, *args, **kwargs)
+            # Commit at the outermost lock level so the SQLite WAL write
+            # lock is surrendered before _WRITER_LOCK is released.  When
+            # writer_lock_held_count()==1 this is the sole holder; when
+            # >1 a caller already holds the lock and owns the transaction
+            # boundary — skip the commit so we don't interrupt it.
+            if writer_lock_held_count() == 1:
+                try:
+                    conn.commit()
+                except Exception:
+                    pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)